cpp/oneapi/dal/algo/basic_statistics/compute_types.hpp

```diff
@@ -0,0 +1,211 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/algo/basic_statistics/common.hpp"
+
+namespace oneapi::dal::basic_statistics {
+
+namespace detail {
+namespace v1 {
+template <typename Task>
+class compute_input_impl;
+
+template <typename Task>
+class compute_result_impl;
+} // namespace v1
+
+using v1::compute_input_impl;
+using v1::compute_result_impl;
+
+} // namespace detail
+
+namespace v1 {
+
+/// @tparam Task Tag-type that specifies the type of the problem to solve. Can
+///              be :expr:`task::v1::compute`.
+template <typename Task = task::by_default>
+class compute_input : public base {
+    static_assert(detail::is_valid_task_v<Task>);
+
+public:
+    using task_t = Task;
+
+    /// Creates a new instance of the class with the given :literal:`x` and :literal:`y`.
+    compute_input(const table& data);
+
+    /// An $n1 \\times p$ table with the data x, where each row
+    /// stores one feature vector.
+    /// @remark default = table{}
+    const table& get_data() const;
+
+    auto& set_data(const table& data) {
+        set_data_impl(data);
+        return *this;
+    }
+
+protected:
+    void set_data_impl(const table& data);
+
+private:
+    dal::detail::pimpl<detail::compute_input_impl<Task>> impl_;
+};
+
+/// @tparam Task Tag-type that specifies the type of the problem to solve. Can
+///              be :expr:`task::v1::compute`.
+template <typename Task = task::by_default>
+class compute_result : public base {
+    static_assert(detail::is_valid_task_v<Task>);
+
+public:
+    using task_t = Task;
+
+    /// Creates a new instance of the class with the default property values.
+    compute_result();
+
+    /// A $n1 \\times n2$ table with the minimum result.
```

Comment:
Misleading doxygen for all functions, should be `n \\times 1`. Need to mention that each j-th element is minimum over all observations for the j-th feature. The same for other results.

Reply:
fixed